core/kerberos/kerberos-core/src/main/java/com/thinkbiganalytics/kerberos/KerberosUtil.java

```diff
@@ -20,45 +20,65 @@
  * #L%
  */
 
+import com.google.common.base.Throwables;
+
 import org.apache.hadoop.security.UserGroupInformation;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.security.PrivilegedExceptionAction;
 import java.sql.Connection;
 import java.sql.SQLException;
+import java.util.concurrent.Callable;
 
+import javax.annotation.Nonnull;
 import javax.sql.DataSource;
 
 /**
+ * Utility methods for executing code that requires an active Kerberos ticket.
  */
 public class KerberosUtil {
 
     private static final Logger log = LoggerFactory.getLogger(KerberosUtil.class);
 
+    /**
+     * Gets a connection for the specified data source using the optional Kerberos ticket.
+     */
     public static Connection getConnectionWithOrWithoutKerberos(final DataSource dataSource, KerberosTicketConfiguration kerberosTicketConfiguration) throws SQLException {
-        Connection connection = null;
-        if (kerberosTicketConfiguration.isKerberosEnabled()) {
-            log.info("Initializing Kerberos ticket for Hive connection");
-            UserGroupInformation userGroupInformation;
-            try {
-                KerberosTicketGenerator t = new KerberosTicketGenerator();
-                userGroupInformation = t.generateKerberosTicket(kerberosTicketConfiguration);
-                connection = userGroupInformation.doAs(new PrivilegedExceptionAction<Connection>() {
-                    @Override
-                    public Connection run() throws Exception {
+        return runWithOrWithoutKerberos(new Callable<Connection>() {
+            @Override
+            public Connection call() throws Exception {
+                try {
+                    return dataSource.getConnection();
+                } catch (final Exception e) {
+                    log.error("Error in Kerberos authentication", e);
+                    throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)